### ISSUE TITLE
BUG: printed_every_n_steps and keep_every cannot be equal

### ISSUE THREAD

**Issue by bangxiangyong** [automl/pybnn#6]
When `print_every_n_steps` and `keep_every` are set to equal, the network is not updated properly

Following the notebook example:
```
model = Bohamiann(print_every_n_steps=50)
model.train(x[:, None], y, num_steps=20000, num_burn_in_steps=2000, keep_every=50, lr=1e-2, verbose=True)
```
results in : 
![bug](https://user-images.githubusercontent.com/8087472/71605409-9bdc9200-2b60-11ea-9f5e-ca168ac3d5cf.png)

The reason is because the seemingly innocent `print_every_n_steps` does an important job in the background : by calling `network_predict`, implicitly the current weights is reset to the latest sampled_weights via the line : `self.network_weights = weights`

Upon removing the line where it `prints` the network performance, we can log and notice the drifting loss weights prior shown as `WP` here:

![growing p_w](https://user-images.githubusercontent.com/8087472/71605650-84060d80-2b62-11ea-9ca6-7092482358d6.JPG)

**Comment by aaronkl** [comment]
good catch thanks, could you  send a PR to fix it?

**Comment by bangxiangyong** [comment]
I don't know how to fix it unfortunately !..
